File: 0682/sesion15 - Master-Detail/SoftInv/SoftInvDBManager/DBManagerMSSQL.cs
using System;
using System.Collections.Generic;
using System.Data.Common;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Microsoft.Data.SqlClient;

namespace SoftInv.Db
{
    public class DBManagerMSSQL: DBManager
    {
        private DBManagerMSSQL()
        {
            //construtor ocultado que no se puedan crear instancias desde fuera de la clase
        }

        public static void CreateConcreteInstance()
        {
            if (DBManager.dbManager == null)
                DBManager.dbManager = new DBManagerMSSQL();
        }

        protected override void RealizarConexionABaseDeDatos(string cadenaDeConexion)
        {
            this.conexion = new SqlConnection(cadenaDeConexion);
        }

        protected override string ObtenerCadenaDeConexion()
        {
            SqlConnectionStringBuilder csBuilder = new SqlConnectionStringBuilder()
            {
                DataSource = $"{this.nombreDeHost},{this.puerto}",
                InitialCatalog = this.baseDeDatos,
                UserID = this.usuario,
                Password = DBManager.Descifrar(this.contraseña),
                TrustServerCertificate = true
            };

            return csBuilder.ConnectionString;            
        }

        public override DbCommand CrearComando()
        {
            return new SqlCommand();
        }

        public override string RetornarSQLParaUltimoAutoGenerado()
        {
            string sql = "select @@IDENTITY as id";
            return sql;
        }
    }
}

File: 0682/sesión 22 - ex1_2025_2_P3/Pregunta 03 - Estudiantes/Pregunta3_ProyectoBase/TransitSoftDBManager/src/main/java/pe/pucp/transitsoft/db/DBManagerMSSQL.java
package pe.pucp.transitsoft.db;

public class DBManagerMSSQL extends DBManager {

    protected DBManagerMSSQL() {
        
    }

    @Override
    protected String getURL() {
        String url = this.tipo_de_driver.concat("://");
        url = url.concat(this.nombre_de_host);
        url = url.concat(":");
        url = url.concat(this.puerto);
        url = url.concat(";");
        url = url.concat("databaseName=" + this.base_de_datos);
        url = url.concat(";encrypt=false");
        System.out.println(url);
        return url;
    }
    
    @Override
    public String retornarSQLParaUltimoAutoGenerado(){
        String sql = "SELECT @@IDENTITY AS ID";
        return sql;
    }
    
}
